src/sagemaker/amazon/amazon_estimator.py

```diff
@@ -64,6 +64,20 @@ def data_location(self, data_location):
             data_location = data_location + '/'
         self._data_location = data_location
 
+    @classmethod
+    def _from_training_job(cls, init_params, hyperparameters, image, sagemaker_session):
+
+        # The hyperparam names may not be the same as the class attribute that holds them,
+        # for instance: local_lloyd_init_method is called local_init_method. We need to map these
+        # and pass the correct name to the constructor.
+
+        for attribute, value in cls.__dict__.items():
+            if isinstance(value, hp):
```

Comment:
Accessing __dict__ to find a class attribute shows that ```_from_training_job``` should not be a class method but an instance method.

Reply:
I think this is fine here, hyperparameters are implemented as descriptor objects.